routes/weatherProviders/DWD.ts

```diff
@@ -0,0 +1,291 @@
+import * as moment from "moment-timezone";
+
+import { GeoCoordinates, WeatherData, ZimmermanWateringData } from "../../types";
+import { httpJSONRequest } from "../weather";
+import { WeatherProvider } from "./WeatherProvider";
+import { approximateSolarRadiation, CloudCoverInfo, EToData } from "../adjustmentMethods/EToAdjustmentMethod";
+import { CodedError, ErrorCode } from "../../errors";
+
+export default class DWDWeatherProvider extends WeatherProvider {
+
+	public constructor() {
+		super();
+	}
+
+	public async getWateringData( coordinates: GeoCoordinates ): Promise< ZimmermanWateringData > {
+		const yesterdayTimestamp: string = moment().subtract( 1, "day" ).utc().format("YYYY-MM-DD[T]hh");
```

Comment:
There's a bug here: Using 'hh' in the time format produces the hours in 12-hour format. This leads to afternoon hours to be misinterpreted. The easiest way to fix this is simply to use toISOString() instead of format(...).